examples/vm_scheduling/reinforcement_learning/common/components/state.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import numpy as np
+import pandas as pd
+
+
+PM_ATTRIBUTES = ["cpu_cores_capacity", "memory_capacity", \
+                 "cpu_cores_allocated", "memory_allocated"]
+
+VM_ATTRIBUTES = ["cpu_cores_requirement", "memory_requirement", "lifetime", "remain_time", "total_income"]
+
+
+class VMState(object):
+    def __init__(
+        self, pm_num, durations, vm_state_path, vm_window_size, pm_window_size
+    ):
+        self._pm_num = pm_num
+        self._durations = durations
+        self._vm_states = np.load(vm_state_path)
+        self._dim = (pm_num * 2) * pm_window_size + len(VM_ATTRIBUTES) * vm_window_size
```

Comment:
Add comment for the PM attributes used (why * 2)

Reply:
I'll add a variable PM_DIM and explain the variable